src/utils/time-utils.ts

```diff
@@ -30,14 +48,15 @@ export const getHours = (minutes: number) =>
  */
 export const formatTimePeriod = (timespan: string[] | undefined) => {
   if (!timespan) return null;
+  const language = useAtomValue(languageAtom);
 
   if (timespan[0].length > 4) {
-    const startDate = new Date(timespan[0].split("-").join(", ")).toLocaleDateString(
-      strings.localization.time
-    );
-    const endDate = new Date(timespan[1].split("-").join(", ")).toLocaleDateString(
-      strings.localization.time
-    );
+    const startDate = DateTime.fromJSDate(new Date(timespan[0]))
+      .setLocale(language)
```

Comment:
Isn't Luxons locale already defined in `App.tsx`

Reply:
Yes, it indeed is. Removed the set locales.